// @cartesia/cartesia-js/serialization/resources/tts/types/OutputFormat.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutputFormat = void 0;
const core = __importStar(require("../../../../core"));
const RawOutputFormat_1 = require("./RawOutputFormat");
const WavOutputFormat_1 = require("./WavOutputFormat");
const Mp3OutputFormat_1 = require("./Mp3OutputFormat");
exports.OutputFormat = core.serialization
    .union("container", {
    raw: RawOutputFormat_1.RawOutputFormat,
    wav: WavOutputFormat_1.WavOutputFormat,
    mp3: Mp3OutputFormat_1.Mp3OutputFormat,
})
    .transform({
    transform: (value) => value,
    untransform: (value) => value,
});
