// @cartesia/cartesia-js/api/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.voices = exports.voiceChanger = exports.tts = exports.embedding = exports.datasets = exports.apiStatus = void 0;
exports.apiStatus = __importStar(require("./apiStatus"));
__exportStar(require("./apiStatus/types"), exports);
exports.datasets = __importStar(require("./datasets"));
__exportStar(require("./datasets/types"), exports);
exports.embedding = __importStar(require("./embedding"));
__exportStar(require("./embedding/types"), exports);
exports.tts = __importStar(require("./tts"));
__exportStar(require("./tts/types"), exports);
exports.voiceChanger = __importStar(require("./voiceChanger"));
__exportStar(require("./voiceChanger/types"), exports);
exports.voices = __importStar(require("./voices"));
__exportStar(require("./voices/types"), exports);
__exportStar(require("./datasets/client/requests"), exports);
__exportStar(require("./voiceChanger/client/requests"), exports);
__exportStar(require("./voices/client/requests"), exports);
